kclvm/runner/src/assembler.rs

```diff
@@ -0,0 +1,162 @@
+use indexmap::IndexMap;
+use std::{
+    collections::HashMap,
+    path::{Path, PathBuf},
+    sync::mpsc::channel,
+};
+use threadpool::ThreadPool;
+
+use crate::command::Command;
+use kclvm_ast::ast;
+use kclvm_compiler::codegen::{llvm::emit_code, EmitOptions};
+use kclvm_config::cache::{load_pkg_cache, save_pkg_cache, CacheOption};
+use kclvm_sema::resolver::scope::ProgramScope;
+
+/// LLVM IR file suffix.
+const LL_FILE: &str = "_a.out";
+
+/// KclvmAssembler is mainly responsible for generating bytecode,
+/// LLVM IR or dylib, and take the result of kclvm-parser and kclvm-sema
```

Comment:
Ultimately the assembler should not be directly aware of llvm.

Reply:
I modified the comments for KclvmAssembler and encapsulated the part that generates the llvm code in static method "lock_file_and_gen_dylib".